src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -111,6 +111,30 @@ internal static void GetUnderlyingTypeChain(NamedTypeSymbol underlyingTupleType,
             }
         }
 
+        /// <summary>
+        /// Gets flattened type arguments of the underlying type
+        /// which correspond to the types of the tuple elements left-to-right
+        /// </summary>
+        internal static void GetElementTypes(NamedTypeSymbol underlyingTupleType, ArrayBuilder<TypeSymbol> tupleElementTypes)
```

Comment:
It feels like this method is not necessary, we should be able to simply ask for all fields from the tuple type symbol and look at their types.